programs/svm-spoke/src/instructions/deposit.rs

```diff
@@ -101,21 +105,27 @@ pub fn _deposit(
         if exclusivity_deadline <= MAX_EXCLUSIVITY_PERIOD_SECONDS {
             exclusivity_deadline += current_time;
         }
-
         if exclusive_relayer == Pubkey::default() {
             return err!(CommonError::InvalidExclusiveRelayer);
         }
     }
 
-    // Depositor must have delegated input_amount to the state PDA.
+    // Verify delegate PDA
+    let (pda, bump) = Pubkey::find_program_address(&[b"delegate", &delegate_seed_hash], &ctx.program_id);
+    if pda != ctx.accounts.delegate.key() {
+        return err!(SvmError::InvalidDelegatePda);
+    }
```

Comment:
better move this logic inside `transfer_from` by passing `delegate_seed_hash` as we have the same logic for fills